lib/node/spvnode.js

```diff
@@ -275,25 +275,37 @@ class SPVNode extends Node {
   }
 
   /**
-   * Broadcast a transaction (note that this will _not_ be verified
-   * by the mempool - use with care, lest you get banned from
-   * bitcoind nodes).
+   * Add transaction to txFilter, broadcast (note that this will
+   * _not_ be verified by the mempool - use with care, lest you
+   * get banned from bitcoind nodes).
    * @param {TX} tx
-   * @returns {Promise}
    */
 
-  sendTX(tx) {
-    return this.broadcast(tx);
+  async sendTX(tx) {
+    try {
+      this.pool.txFilter.added(tx.hash());
+      this.emit('tx', tx);
+    } catch (e) {
+      this.emit('error', e);
+    }
+
+    // We need to announce by hand since
+    // spv nodes are always selfish
+    await this.broadcast(tx);
+    return;
   }
 
   /**
    * Broadcast a transaction. Silence errors.
    * @param {TX} tx
-   * @returns {Promise}
```

Comment:
you still need this returns in both methods.

Reply:
I can do it, but won't this make the spv node have a different API from the full node? Is there a reason for this difference?

Reply:
fullnode lacks this line in docs. Every `async` method returns `Promise`.